build/Dockerfile

```diff
@@ -131,52 +133,85 @@ apt-get update
 apt-get install -y --no-install-recommends \
     ca-certificates \
     curl \
-    git \
-    jq \
-    wget \
-    xxd
+    git
 EOF
 
 # Install Foundry from downloaded pre-compiled binaries
 ARG FOUNDRY_NIGHTLY_VERSION
 RUN curl -sSL https://github.com/foundry-rs/foundry/releases/download/nightly-${FOUNDRY_NIGHTLY_VERSION}/foundry_nightly_linux_$(dpkg --print-architecture).tar.gz | \
     tar -zx -C /usr/local/bin
 
+
 # STAGE: devnet-builder
 #
 # This stage builds the devnet state that will be loaded in Anvil.
 FROM devnet-base as devnet-builder
 ARG DEVNET_BUILD_PATH
 WORKDIR ${DEVNET_BUILD_PATH}
 
+# Install dependencies needed to install yarn
+ARG DEBIAN_FRONTEND=noninteractive
+RUN <<EOF
+set -e
+apt-get update
+apt-get install -y --no-install-recommends \
+    gnupg
+EOF
+
+# Install yarn
+ARG DEBIAN_FRONTEND=noninteractive
+RUN <<EOF
+set -e
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+apt-get update
+apt-get install -y --no-install-recommends \
+    nodejs \
+    yarn
+EOF
+
+# Copy scripts
+COPY . .
```

Comment:
We should avoid copying everything because it will invalidate the docker cache. Especially things like the .git directory.

Reply:
It looks like we're doing the same with the node build at https://github.com/cartesi/rollups-node/blob/4c74729249cebd4b0c74e4d4bafd103f96d7eaef/build/Dockerfile#L245.

So, I'll:

- adjust `.gitignore` to not copy any git/github related file; and
- restrict the copied directories to the ones needed to build devnet

Reply:
Don't confuse `.gitignore` with `.dockerignore`.

Reply:
Changes made at af52782a and ee968dc7